### DISCUSSION TITLE
Example for Next.js with Gitpod (Cloud IDE) and working Fast Refresh

### DISCUSSION THREAD

**Discussion by medihack** [vercel/next.js#22522] [Ideas]
**Describe the feature you'd like to request**

Cloud IDEs like [Gitpod](https://www.gitpod.io/) or [Github Workspaces](https://github.com/features/codespaces) will get a big role in the future. I tried out Next.js with Gitpod, but can't make Fast Refresh work. The [recommended solutions] do not work for Next.js (the Next.js dev server does not have those options). I put up a little test project at https://github.com/medihack/gitpod-next (just click on the Gitpod link). The connection to https://3000-plum-jellyfish-uk215sfw.ws-eu03.gitpod.io/_next/webpack-hmr?page=/ fails (504).

**Describe the solution you'd like**

An example of a setup with Gitpod and working fast refresh.

**Describe alternatives you've considered**

Not using Cloud IDEs.

**Comment by abeauvois** [comment]
I've solved the hmr reloding because not accessing to the server via wss://... by adding this vite.config.js options:

``` json
// https://vitejs.dev/config/
export default defineConfig({
  https: true,
  server: {
    host: true,
    port: 3000,
    strictPort: true,
    hmr: {
      port: 443,
    }
  },
...
})```
